utils/build-script

```diff
@@ -1040,455 +1513,37 @@ details of the setups of other systems or automated environments.""")
     if args.cmake is not None:
         toolchain.cmake = args.cmake
 
-    if toolchain.cc is None or toolchain.cxx is None:
-        diagnostics.fatal(
-            "can't find clang (please install clang-3.5 or a later version)")
+    # Validate the arguments.
+    BuildScriptInvocation.validate_arguments(toolchain, args)
 
-    if toolchain.cmake is None:
-        diagnostics.fatal("can't find CMake (please install CMake)")
+    # Preprocess the arguments to apply defaults.
+    BuildScriptInvocation.apply_default_arguments(toolchain, args)
```

Comment:
I agree.
My concern is, in the future, `apply_default_arguments ` would use `toolchain.{cc,cxx,cmake}`.
Maybe, we should split `validate_arguments` into two parts: before `apply_default_arguments` and after that.